src/template/template.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { ResetPasswordEmailData } from './dto/reset-password-email-data.dto';
+import { readFileSync } from 'fs';
+import * as Handlebars from 'handlebars';
+import mjml2html = require('mjml');
+
+@Injectable()
+export class TemplateService{
+  private readonly templatesFolderPath = './src/template/templates/';
+
+  getResetPasswordEmail(data: ResetPasswordEmailData): string {
+    const template = this._getResetPasswordEmailTemplate();
+    return this._compileAndInjectData(template, data);
+  }
+
+  protected _getResetPasswordEmailTemplate(): string {
+    return readFileSync(`${this.templatesFolderPath}reset-password.mjml`, 'utf8');
+  }
+
+  protected _compileAndInjectData(mjmlTemplate: string, data: ResetPasswordEmailData): string {
```

Comment:
Will TemplateService only be used for ResetPassword? If not, this needs to be made generic. @katafalque @mfozmen